File: src/main/java/org/dama/datasynth/runtime/spark/operators/matching/graphs/types/GraphPartitioner.java
package org.dama.datasynth.runtime.spark.operators.matching.graphs.types;

/**
 * Created by aprat on 27/02/17.
 */
public abstract class GraphPartitioner {

    protected  Graph graph = null;
    protected Traversal traversal = null;
    public GraphPartitioner(Graph graph, Class<? extends Traversal> traversalType) {
        this.graph = graph;
        try {
            traversal = traversalType.newInstance();
            traversal.initialize(this.graph);
        } catch ( Exception e) {
            throw new RuntimeException(e);
        }
    }

    public abstract Partition getPartition();
}

File: src/main/scala/org/dama/datasynth/runtime/spark/operators/matching/GraphPartitioner.scala
package org.dama.datasynth.runtime.spark.operators.matching


import org.apache.spark.sql.DataFrame
import org.dama.datasynth.executionplan.ExecutionPlan.EdgeTable
import org.dama.datasynth.runtime.spark.operators.matching.utils.traversals.Traversal
import org.dama.datasynth.runtime.spark.operators.matching.utils.Graph
/**
  * Created by joangui on 17/07/2017.
  */
abstract class GraphPartitioner(edgeDataframe: DataFrame, traversal:Traversal) {
  val graph:Graph = new Graph(edgeDataframe)
  traversal
}
